src/go/MONIT/ggus_alerting.go

```diff
@@ -0,0 +1,332 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"os"
+	"strconv"
+	"time"
+)
+
+// File       : ggus_alerting.go
+// Author     : Rahul Indra <indrarahul2013 AT gmail dot com>
+// Created    : Thu, 21 May 2020 21:07:32 GMT
+// Description: GGUS Alerting Module for CERN MONIT infrastructure
+
+//URL for AlertManager
+var alertManagerURL string
+
+//severity of alerts
+var severity string
+
+//GGUS tag
+var tag string
+
+//Required VO attribute
+var vo string
+
+//verbose defines verbosity level
+var verbose int
+
+//Map for storing Existing Tickets
+var exstTkt map[string]int
+
+//GGUS Data Struct
+type ggus []struct {
+	TicketID        int    `json:"TicketID"`
+	Type            string `json:"Type"`
+	VO              string `json:"VO"`
+	Site            string `json:"Site"`
+	Priority        string `json:"Priority"`
+	ResponsibleUnit string `json:"ResponsibleUnit"`
+	Status          string `json:"Status"`
+	LastUpdate      int    `json:"LastUpdate"`
+	Subject         string `json:"Subject"`
+	Scope           string `json:"Scope"`
+}
+
+//AlertManager API acceptable JSON Data for GGUS Data
+type amJSON struct {
+	Labels struct {
+		Alertname string `json:"alertname"`
+		Severity  string `json:"severity"`
+		Tag       string `json:"tag"`
+	} `json:"labels"`
+	Annotations struct {
+		Priority        string `json:"Priority"`
+		ResponsibleUnit string `json:"ResponsibleUnit"`
+		Scope           string `json:"Scope"`
+		Site            string `json:"Site"`
+		Status          string `json:"Status"`
+		Subject         string `json:"Subject"`
+		TicketID        string `json:"TicketID"`
+		Type            string `json:"Type"`
+		VO              string `json:"VO"`
+		URL             string `json:"URL"`
+	} `json:"annotations"`
+	StartsAt time.Time `json:"startsAt"`
+	EndsAt   time.Time `json:"endsAt"`
+}
+
+//AlertManager GET API acceptable JSON Data struct for GGUS data
+type ggusData struct {
+	Data []amJSON
```

Comment:
If ggusData is really a list of amJSON then why do you need this struct at all? You can easily Unmarshal data using `[]amJSON`.

Reply:
Valentin,
When I call /api/v1/alerts (GET) API, I get struct of data array from AlertManager. It is of form :- 
```
{
   "data" : [ {},
              {},
             ],
. . . 
}
```

So, I would be requiring struct to parse the information.

Reply:
ok, then it is correct.